test/udp_backend_test.rb

```diff
@@ -132,4 +132,16 @@ def test_io_error_should_not_raise_but_log
     @logger.expects(:error)
     StatsD.increment('fail')
   end
+
+  def test_thread_error_does_not_raise
```

Comment:
It's a test that nothing raises (and exits cleanly.) Acquiring a lock when exiting raises a `ThreadError`, so that's what I'm rescuing.